test/minikube/minikube_priority_class_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build long
+// +build long
+
+package minikube
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/nuodb/nuodb-helm-charts/v3/test/testlib"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+)
+
+func TestPriorityClassNonexistent(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+
+	randomSuffix := strings.ToLower(random.UniqueId())
+	namespaceName := fmt.Sprintf("%skubernetespriorityclassnonexistent-%s", testlib.NAMESPACE_NAME_PREFIX, randomSuffix)
+	testlib.CreateNamespace(t, namespaceName)
+
+	// create a database
+	priorityClass := "high-priority"
+	options := helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    testlib.MINIMAL_VIABLE_ENGINE_CPU,
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.sm.noHotCopy.replicas":        "1",
+			"admin.priorityClass":                   priorityClass,
+			"database.priorityClasses.sm":           priorityClass,
+			"database.priorityClasses.te":           priorityClass,
+		},
+	}
+
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+	helmChartReleaseName, _ := testlib.StartAdminNoWait(t, &options, 1, namespaceName)
+	adminStatefulSet := fmt.Sprintf("%s-nuodb-cluster0", helmChartReleaseName)
+	admin0 := adminStatefulSet + "-0"
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE)
+	testlib.StartDatabaseNoWait(t, namespaceName, admin0, &options)
+	testlib.Await(t, func() bool {
+		filter := metav1.ListOptions{FieldSelector: "type=Warning"}
+		events := testlib.GetEvents(t, namespaceName, filter)
+		for _, event := range events {
+			if !strings.Contains(event.Message, "no PriorityClass with name high-priority was found") {
+				return false
```

Comment:
Am I understanding this correctly? Returning `false` inside the loop means that we should have _only_ events with the expected string. Can we have other warnings in the system?

Maybe it's better to count all occurrences of the expected event and wait until they become 4.

Reply:
Events are namespace-bound, so I wouldn't expect there to be any events of type `Warning` that are not generated by the failure to create pods. I could make the check more specific to account for that and also to check that the event is associated with a pod from one of our StatefulSets and Deployments and has reason `FailedCreate`.

Reply:
Thanks for updating the test case!